march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -316,4 +339,18 @@ void MarchHardwareInterface::updatePowerNet()
     }
   }
 }
+
+void MarchHardwareInterface::updateIMotionCubeState(int jointIndex)
+{
+    std::vector<std::string> IMotionCubeState = marchRobot.getJoint(joint_names_[jointIndex]).getIMotionCubeState();
+
+    realtime_pubs_->msg_.joint_names.push_back(joint_names_[jointIndex]);
+    realtime_pubs_->msg_.status_word.push_back(IMotionCubeState[0]);
+    realtime_pubs_->msg_.detailed_error.push_back(IMotionCubeState[1]);
+    realtime_pubs_->msg_.motion_error.push_back(IMotionCubeState[2]);
+    realtime_pubs_->msg_.state.push_back(IMotionCubeState[3]);
+    realtime_pubs_->msg_.detailed_error_description.push_back(IMotionCubeState[4]);
+    realtime_pubs_->msg_.motion_error_description.push_back(IMotionCubeState[5]);
```

Comment:
Here you see why a custom object is desirable. There is nothing that guarantees this order is correct